### ISSUE TITLE
Unexpected Exception on delete key 

### ISSUE THREAD

**Issue by skedus** [ragardner/tksheet#61]
Hi
I found an unexpected exception on delete by key or by the popup menu 'Del'  (Ctrl-X is working)

```
Exception in Tkinter callback
Traceback (most recent call last):
  File "...Programs\Python\Python38\lib\tkinter\__init__.py", line 1883, in __call__
    return self.func(*args)
  File "...\tksheet\_tksheet_main_table.py", line 696, in delete_key
    boxes.append((box, "cells"))
UnboundLocalError: local variable 'boxes' referenced before assignment
```
Below a small example to reproduce it:

```
import tkinter as tk
from tksheet import Sheet

root = tk.Tk()
frame=tk.Frame(root) 

data1 = [
    ['value11','value12','value13','value14'],
    ['value21','value22','value23','value24'],
    ['value31','value32','value33','value34'],
]

sheet1 = Sheet(frame, data=data1, height = 150)
sheet1.enable_bindings(
    ("single_select",
    "arrowkeys",
    "column_width_resize",
    "right_click_popup_menu",
    "copy",
    "cut",
    "paste",
    "delete",
    "edit_cell")
)

sheet1.pack(expand=tk.YES, fill=tk.BOTH)
frame.pack(fill=tk.BOTH, expand=tk.YES)

root.mainloop()
```

**Comment by ragardner** [comment]
Thanks for the report and example, it's hopefully fixed in `5.0.12`

**Comment by skedus** [comment]
I tested the v5.0.12 this is fine thank you,
I can now use the extra binding (end_delete) really good :-)
Thank you
